### PR TITLE
 Add link, clarifying justification for orphaned role tests

### PR DESCRIPTION
**Pull request by nmlapre**
### UPDATE
This PR adds comments explaining the purpose of and justification for the orphaned role tests in various files. This comment includes a link to the relevant piece of the CORE-AAM specification.

The previous version of this PR moved the orphaned tests. See that description below.

---
### Original Intent

Closes https://github.com/web-platform-tests/interop-accessibility/issues/109

See longer description of the issue: https://github.com/web-platform-tests/interop-accessibility/issues/109
See test change proposal issue: https://github.com/web-platform-tests/interop/issues/645
See related ARIA issue: https://github.com/w3c/aria/issues/2137

This PR moves disputed/invalid orphaned role tests to a unified tentative test file. These cases are considered author errors, which, as of the current ARIA spec language, is explicitly not the responsibility of user agents to remedy.

This PR does move some tests that weren't previously running in their source files (due to lack of `verify[Generic]RolesBySelector` calls). I presume those were meant to be running. Either way, they suffer from the same issues and should be grouped in the tentative file.

By my count, this should remove 17 currently-running tests from WPT. All tested browsers are currently failing these tests, so I expect a ~1.4% increase in the accessibility focus area score across the board when this merges.

This is my first non-automated WPT PR, so please assign reviewers as needed. Thanks!

### CODE REVIEW

**Comment by cookiecrook** [comment]
Requesting @scottaohara's review in case there are some orphaning rules in HTML-AAM that we need to keep tests for that I'm not remembering. I'll do my best to review in the context of the ARIA Core spec.

**Comment by cookiecrook** [comment]
And thanks for posting this PR!

File: wai-aria/role/list-roles.html
Comment (cookiecrook) [inline wai-aria/role/list-roles.html:21]: Good catch that this one was not running due to the lack of `verifyGenericRoleByselector();`!

File: wai-aria/role/tab-roles.html
Comment (cookiecrook) [inline wai-aria/role/tab-roles.html:87]: Good catch that this one was not running due to the lack of `verifyGenericRoleByselector();`!

**Review by cookiecrook — COMMENTED** [review]
Looks okay to me once @scottaohara approves for HTML-AAM.

**Review by cookiecrook — CHANGES_REQUESTED** [review]
Hang on.... there are some relevant places in Core-AAM...

> When an element has a role but is not contained in the required context (for example, an orphaned listitem without the required accessible parent of role list), User Agents MUST ignore the role token, and return the computedrole as if the ignored role token had not been included.

```html
<div role="listitem"> <!-- Author error: orphaned listitem. computedrole returns "generic" -->

<div role="list"> <!-- computedrole returns "list" -->
  <div role="listitem"> <!-- computedrole returns "listitem" in the required context. -->
```

Cite: https://w3c.github.io/core-aam/#roleMappingComputedRole

So this needs more discussion and consideration before landing.

**Comment by cookiecrook** [comment]
@twilco brought up a related case where the list markup is invalid, but one or more engines still render the bullet, and therefore may need an exception in the spec, if the implementations agree.

```html
<ul>
  <div>
    <li>one</li>
  </div>
  <li>two</li>
  <li>three</li>
</ul>
```

And I'm still looking for an RFC-2119 statement that covers the native containment case without an explicitly-defined role attr.

**Comment by nmlapre** [comment]
> Hang on.... there are some relevant places in Core-AAM...

Ah! Thanks for finding that. That language seems pretty cut-and-dry to me.

I suppose I'm wondering who wins out between the ARIA's "authors MUST" / "user agents are not responsible" language and CORE-AAM's "user agents MUST" wording. Does that tension require a spec clarification? Either way, I now think this PR is probably not the right thing to do.

**Comment by cookiecrook** [comment]
Yeah, there is still some spec clean-up to do, but I agree the PR can't land as is. Still looking for some other cross-references between other meetings. 

If nothing else, the PR could just add a link to Core-AAM from near the orphaned case, so it will be obvious to the next engineer who would otherwise ask about it.

**Comment by cookiecrook** [comment]
> That language seems pretty cut-and-dry to me.

And FWIW, if there is a significant implementation challenge or cost, we can still raise that for spec reconsideration.

**Comment by nmlapre** [comment]
> If nothing else, the PR could just add a commented link to that section of Core-AAM near the orphaned case, so it will be obvious to the next engineer who would otherwise ask about it.

Alrighty, done! Updated this PR with comments to that effect.

> And FWIW, if there is a significant implementation challenge or cost, we can still raise that for spec reconsideration.

The patch for this in Gecko was simple enough to write, but it's definitely more tree walking. I haven't profiled it to see whether it's a serious performance problem. See my comment in https://github.com/w3c/aria/issues/2137#issuecomment-1989213362.

File: wai-aria/role/list-roles.html
Comment (cookiecrook) [reply wai-aria/role/list-roles.html:21]: Just a reminder that your prior PR fixed (in a new file) the lack of `verifyGenericRoleByselector()` in list-roles.html and tab-roles.html... You can add those back in here if you like, or we can handle it in:
- https://github.com/web-platform-tests/interop-accessibility/issues/111. 

Either is fine with me. Thanks,

**Review by cookiecrook — APPROVED** [review]

File: wai-aria/role/list-roles.html
Comment (nmlapre) [reply wai-aria/role/list-roles.html:21]: Definitely, thank you! I think I'll handle it in https://github.com/web-platform-tests/interop-accessibility/issues/111, just to separate things a bit between (1) resolution of spec question and (2) enabling tests.

Comment (nmlapre) [reply wai-aria/role/list-roles.html:21]: Addressed (2) in the PR here: https://github.com/web-platform-tests/wpt/pull/45174

**Pull request merged**
